pandas/io/sql.py

```diff
@@ -837,11 +837,17 @@ def _query_iterator(
         self, result, chunksize, columns, coerce_float=True, parse_dates=None
     ):
         """Return generator through chunked result set."""
+        has_read_data = False
         while True:
             data = result.fetchmany(chunksize)
             if not data:
+                if not has_read_data:
+                    yield DataFrame.from_records(
+                        [], columns=columns, coerce_float=coerce_float
+                    )
                 break
```

Comment:
Is this `break` statement still being hit?

Reply:
Yes, if `has_read_data` an empty dataframe is yielded, then the `break` statement breaks out of the `while True` loop, ending the generator.

You can verify that this statement is hit by running for example

```
> pytest --cov=pandas/io/ --cov-report html pandas/tests/io/test_sql.py::TestSQLApi::test_read_sql_with_chunksize_no_result
```